packages/database/src/entities/AppConfig.ts

```diff
@@ -0,0 +1,172 @@
+    super(extraValues);
+
+    this.key = key;
+    this.value = value;
+  }
+
+  getSafeKeys(): AppConfigPropertyKeys[] {
+    return ['key', 'value'];
+  }
+
+  /**
+   * Get the value of a Config item that is a string, and return null if it's not
+   */
+  static async getValueAs(
+    em: EntityManager,
+    key: ConfigKey,
+    valueType: 'string',
+    shouldThrow: false,
+  ): Promise<string | null>;
+
+  /**
+   * Get the value of a Config item that is a string, and throw an error if it's not
+   */
+  static async getValueAs(
+    em: EntityManager,
+    key: ConfigKey,
+    valueType: 'string',
+    shouldThrow: true,
+  ): Promise<string>;
+
+  /**
+   * Get the value of a Config item that is a boolean, and return null if it's not
+   */
+  static async getValueAs(
+    em: EntityManager,
+    key: ConfigKey,
+    valueType: 'boolean',
+    shouldThrow: false,
+  ): Promise<boolean | null>;
+
+  /**
+   * Get the value of a Config item that is a boolean, and throw an error if it's not
+   */
+  static async getValueAs(
+    em: EntityManager,
+    key: ConfigKey,
+    valueType: 'boolean',
+    shouldThrow: true,
+  ): Promise<boolean>;
+
+  /**
+   * Get the value of a Config item that is a number, and return null if it's not
+   */
+  static async getValueAs(
+    em: EntityManager,
+    key: ConfigKey,
+    valueType: 'number',
+    shouldThrow: false,
+  ): Promise<number | null>;
+
+  /**
+   * Get the value of a Config item that is a number, and throw an error if it's not
+   */
+  static async getValueAs(
+    em: EntityManager,
+    key: ConfigKey,
+    valueType: 'number',
+    shouldThrow: true,
+  ): Promise<number>;
+
+  /**
+   * Get the value of a Config item that is a object, and return null if it's not
+   */
+  static async getValueAs(
+    em: EntityManager,
+    key: ConfigKey,
+    valueType: 'object',
+    shouldThrow: false,
+  ): Promise<Record<string, any> | null>;
+
+  /**
+   * Get the value of a Config item that is a object, and throw an error if it's not
+   */
+  static async getValueAs(
+    em: EntityManager,
+    key: ConfigKey,
+    valueType: 'object',
+    shouldThrow: true,
+  ): Promise<Record<string, any>>;
+
+  /**
+   * Get the value of a Config item that is a array, and return null if it's not
+   */
+  static async getValueAs(
+    em: EntityManager,
+    key: ConfigKey,
+    valueType: 'array',
+    shouldThrow: false,
+  ): Promise<any[] | null>;
+
+  /**
+   * Get the value of a Config item that is a array, and throw an error if it's not
+   */
+  static async getValueAs(
+    em: EntityManager,
+    key: ConfigKey,
+    valueType: 'array',
+    shouldThrow: true,
+  ): Promise<any[]>;
+
+  static async getValueAs(
+    em: EntityManager,
+    key: ConfigKey,
+    valueType: 'string' | 'boolean' | 'number' | 'object' | 'array',
+    shouldThrow: boolean,
+  ): Promise<string | boolean | number | Record<string, any> | null> {
```

Comment:
Let me work with you on the types here; we can probably get rid of all the overrides above if we tweak this method a bit